kratos/includes/model_part.h

```diff
@@ -862,7 +862,7 @@ class KRATOS_API(KRATOS_CORE) ModelPart : public DataValueContainer, public Flag
      * @param MeshIndex The Id of the mesh (0 by default)
      * @return The desired properties (pointer)
      */
-    PropertiesType::Pointer pGetProperties(IndexType PropertiesId, IndexType MeshIndex = 0) const;
+    const PropertiesType::Pointer pGetProperties(IndexType PropertiesId, IndexType MeshIndex = 0) const;
```

Comment:
`const PropertiesType::Pointer` is not really const, the object it accesses is still mutable. See an example here: https://godbolt.org/z/fTx97hand.


If you trully want to make it const, it should be
```c++
shared_ptr<const PropertiesType> pGetProperties(...)
```
unfortunatelly kratos does not have a `::ConstPointer` typedef.